app/components/ConnectButton.tsx

```diff
@@ -1,93 +1,76 @@
+}
+
+// https://www.rainbowkit.com/docs/custom-connect-button
+function CustomConnectButton() {
+  useSanctionScreening();
 
   return (
     <ConnectButton.Custom>
       {({ account, chain, openAccountModal, openChainModal, openConnectModal, authenticationStatus, mounted }) => {
-        if (isSanctionedAddress === true) {
-          authenticationStatus = "unauthenticated";
-        }
-
         const ready = mounted && authenticationStatus !== "loading";
         const connected =
           ready && account && chain && (!authenticationStatus || authenticationStatus === "authenticated");
 
-        return (
-          <div
-            {...(!ready && {
-              "aria-hidden": true,
-              style: {
-                opacity: 0,
-                pointerEvents: "none",
-                userSelect: "none",
-              },
-            })}
-          >
-            {(() => {
-              if (authenticationStatus === "unauthenticated") {
-                return <Button onClick={openConnectModal}>Sanctioned Address</Button>;
-              } else if (!connected) {
-                return (
-                  <Button variant="gradient" onClick={openConnectModal}>
-                    Connect Wallet
-                  </Button>
-                );
-              }
+        if (!ready) {
+          // The docs make this a blank state with opactiy 0, but might be better to show a loading indicator of some sort
+          return (
+            <Button variant="gradient" className="pointer-events-none select-none opacity-0">
+              Connect Wallet
+            </Button>
+          );
+        }
 
-              if (chain.unsupported || chain.name == "Ethereum") {
-                return (
-                  <Button variant="danger" onClick={openChainModal}>
-                    Switch to Polygon
-                  </Button>
-                );
-              }
+        if (!connected) {
+          return (
+            <Button variant="gradient" onClick={openConnectModal}>
+              Connect Wallet
+            </Button>
+          );
+        }
 
-              return (
-                <div>
-                  <Button onClick={openAccountModal} variant="gradient">
-                    <div className="flex flex-row gap-2 items-center align-middle">
-                      {account.ensAvatar ? (
-                        account.ensAvatar
-                      ) : (
-                        <div className="rounded-full">
-                          <Jazzicon diameter={20} seed={jsNumberForAddress(account.address)} />
-                        </div>
-                      )}
-                      {account.ensName ? account.ensName : account.displayName}
-                      {/* //TODO: later we will show metric and xmetric balances
+        if (chain.unsupported) {
+          return (
+            <Button variant="danger" onClick={openChainModal}>
+              Switch Chain
```

Comment:
I know they wanted this to specifically say "Switch to Polygon" for now, doesn't matter that much though